Replace Travis CI with CircleCI

arlina-espinoza opened the pull request:
This PR replaces the Travis CI integration with CircleCI. Currently only runs the code sniffer - the same job that is running in travis (kernel, functional and JS tests are commented out for now).
To do after this PR: Enable kernel kernel, functional and JS tests - fix any tests that aren't passing.

cnovak commented on phpcs.xml.dist:
```diff
@@ -18,5 +18,15 @@
+    <rule ref="Drupal.Commenting.FunctionComment.TypeHintMissing">
+      <severity>0</severity>
+    </rule>
+    <rule ref="Drupal.Commenting.GenderNeutralComment.GenderNeutral">
```
I believe we fixed all comments to conform to this standard, I would rather fix comment than add this.

arlina-espinoza replied:
Re-enabled the `TypeHintMissing` and `GenderNeutralComment` sniffs, and yes, they were passing for the module. :)

cnovak commented on .circleci/code-sniffer.sh:
```diff
@@ -0,0 +1,14 @@
+vendor/bin/phpcs --config-set installed_paths vendor/drupal/coder/coder_sniffer
+
+# Check coding standards
+vendor/bin/phpcs -p -s -n --colors --standard=modules/apigee_edge/phpcs.xml.dist --report=junit --report-junit=artifacts/phpcs/phpcs.xml modules/$1
```
Do we want to [keep phpmd and phpmetrics](https://github.com/deviantintegral/drupal_tests/blob/master/hooks/code-sniffer.sh#L13)?

arlina-espinoza replied:
I've re-enabled phpmd and phpmetrics, which will not throw errors, but give interesting insights into the project. For example:
- https://76-174000098-gh.circle-artifacts.com/0/var/www/html/artifacts/phpmd/index.html
- https://76-174000098-gh.circle-artifacts.com/0/var/www/html/artifacts/phpmetrics/index.html

arlina-espinoza commented:
> Woah this is a major DX win @arlina-espinoza Thank you. LGTM +1
> 
> Do we have follow up tickets for "To do after this PR: Enable kernel kernel, functional and JS tests - fix any tests that aren't passing."?

Added follow up issue to enable other tests here: https://github.com/apigee/apigee-edge-drupal/issues/341

@cnovak I've addressed the requested changes, could you review again, please?
